scripts/ResourceAutomation/automationAccount.bicep

```diff
@@ -0,0 +1,216 @@
+    timeZone: scheduleTimezone
+  }
+}
+
+resource weekdaysEndOfDaySchedule 'Microsoft.Automation/automationAccounts/schedules@2022-08-08' = {
+  name: 'weekdays-end-of-day'
+  parent: automationAccount
+  properties: {
+    advancedSchedule: {
+      weekDays: [ 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday' ]
+    }
+    description: 'Schedule that runs every weekday at specified time'
+    interval: 1
+    frequency: 'Week'
+    startTime: scheduleEndOfDayDateTime
+    expiryTime: '9999-12-31T17:59:00-06:00'
+    timeZone: scheduleTimezone
+  }
+}
+
+resource resourceBringupRunbooks 'Microsoft.Automation/automationAccounts/runbooks@2022-08-08' = [for resourceToAutomate in resourcesToAutomate: {
+  name: 'runbook-${resourceToAutomate.clusterName}-bringup'
+  location: location
+  parent: automationAccount
+  properties: {
+    description: 'Runbook to bring up ${resourceToAutomate.clusterName} and ${resourceToAutomate.gatewayName} at the beginning of the day'
+    logProgress: logProgress
+    logVerbose: logVerbose
+    publishContentLink: {
+      uri: clusterGwStartStopRunbookURL
+    }
+    runbookType: 'PowerShell'
+  }
+}]
+
+resource resourceBringupRunbookSchedules 'Microsoft.Automation/automationAccounts/jobSchedules@2022-08-08' = [for resourceToAutomate in resourcesToAutomate: {
+  name: guid('resource-bringup-schedule',resourceToAutomate.clusterName,AA_Name,'bringup')
+  parent: automationAccount
+  properties: {
+    parameters: {
+      tenantId: tenant().tenantId
+      subscriptionName: subscription().displayName
+      automationAccountResourceGroup: resourceGroup().name
+      automationAccountName: AA_Name
+      managedIdentityClientId: automationMI.properties.clientId
+      resourceGroup: resourceToAutomate.resourceGroup
+      clusterName: resourceToAutomate.clusterName
+      gatewayName: resourceToAutomate.gatewayName
+      operation: 'start'
+    }
+    runbook: {
+      name: 'runbook-${resourceToAutomate.clusterName}-bringup'
+    }
+    schedule: {
+      name: 'weekdays-start-of-day'
+    }
+  }
+  dependsOn:[resourceBringupRunbooks,weekdaysStartOfDaySchedule]
+}]
+
+resource resourceShutdownRunbooks 'Microsoft.Automation/automationAccounts/runbooks@2022-08-08' = [for resourceToAutomate in resourcesToAutomate: {
+  name: 'runbook-${resourceToAutomate.clusterName}-shutdown'
+  location: location
+  parent: automationAccount
+  properties: {
+    description: 'Runbook to shut down ${resourceToAutomate.clusterName} and ${resourceToAutomate.gatewayName} at the end of the day'
+    logProgress: logProgress
+    logVerbose: logVerbose
+    publishContentLink: {
+      uri: clusterGwStartStopRunbookURL
+    }
+    runbookType: 'PowerShell'
+  }
+}]
+
+resource resourceShutdownRunbookSchedules 'Microsoft.Automation/automationAccounts/jobSchedules@2022-08-08' = [for resourceToAutomate in resourcesToAutomate: {
+  name: guid('resource-shutdown-schedule',resourceToAutomate.clusterName,AA_Name,'shutdown')
```

Comment:
this is not wrong in your code but there is an issue with conflicts that happen whenever this resource is redeployed. something along the lines of duplicate jobschedule id. I've observed this happen even if we delete resource group containing this resource and then redeploy. non deterministic field is not permitted here to get around it. this would be good feedback to product group